From 770dc46bfc33bd450cc94d3c7fcf423836919933 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Wed, 17 Apr 2024 14:15:06 +0200
Subject: [PATCH] Fix a build issue for PRESCOTT on x86_64
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Define everything the same way for PRESCOTT as for PENTIUM4.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
---
 common_x86.h                            | 2 +-
 kernel/x86_64/dgemm_ncopy_4.S           | 2 +-
 kernel/x86_64/dgemm_tcopy_2.S           | 2 +-
 kernel/x86_64/dgemm_tcopy_4.S           | 2 +-
 kernel/x86_64/gemm_kernel_8x4_sse.S     | 8 ++++----
 kernel/x86_64/gemm_ncopy_4.S            | 2 +-
 kernel/x86_64/gemm_tcopy_4.S            | 8 ++++----
 kernel/x86_64/izamax_sse2.S             | 2 +-
 kernel/x86_64/symv_L_sse.S              | 2 +-
 kernel/x86_64/symv_L_sse2.S             | 2 +-
 kernel/x86_64/symv_U_sse.S              | 2 +-
 kernel/x86_64/symv_U_sse2.S             | 2 +-
 kernel/x86_64/trsm_kernel_LN_8x4_sse.S  | 2 +-
 kernel/x86_64/trsm_kernel_LT_8x4_sse.S  | 2 +-
 kernel/x86_64/trsm_kernel_RT_8x4_sse.S  | 2 +-
 kernel/x86_64/zasum_sse2.S              | 2 +-
 kernel/x86_64/zgemm3m_kernel_8x4_sse.S  | 8 ++++----
 kernel/x86_64/zgemm_kernel_4x2_sse.S    | 4 ++--
 kernel/x86_64/zgemm_ncopy_2.S           | 8 ++++----
 kernel/x86_64/zgemm_tcopy_2.S           | 8 ++++----
 kernel/x86_64/zsum_sse2.S               | 2 +-
 kernel/x86_64/zsymv_L_sse.S             | 2 +-
 kernel/x86_64/zsymv_L_sse2.S            | 2 +-
 kernel/x86_64/zsymv_U_sse.S             | 2 +-
 kernel/x86_64/zsymv_U_sse2.S            | 2 +-
 kernel/x86_64/ztrsm_kernel_LN_4x2_sse.S | 2 +-
 kernel/x86_64/ztrsm_kernel_LT_4x2_sse.S | 2 +-
 kernel/x86_64/ztrsm_kernel_RT_4x2_sse.S | 2 +-
 l1param.h                               | 2 +-
 l2param.h                               | 2 +-
 30 files changed, 46 insertions(+), 46 deletions(-)

diff --git a/common_x86.h b/common_x86.h
index 65fb9a460..c049874c6 100644
--- a/common_x86.h
+++ b/common_x86.h
@@ -229,7 +229,7 @@ static __inline int blas_quickdivide(unsigned int x, unsigned int y){
 #define EMMS
 #endif
 
-#if defined(CORE2) || defined(PENTIUM4)
+#if defined(CORE2) || defined(PENTIUM4) || defined(PRESCOTT)
 #define movapd	movaps
 #endif
 
diff --git a/kernel/x86_64/dgemm_ncopy_4.S b/kernel/x86_64/dgemm_ncopy_4.S
index 41eac9597..4e07c0215 100644
--- a/kernel/x86_64/dgemm_ncopy_4.S
+++ b/kernel/x86_64/dgemm_ncopy_4.S
@@ -39,7 +39,7 @@
 #define ASSEMBLER
 #include "common.h"
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 #define PREFETCHSIZE	16
 #define PREFETCH      prefetcht0
 #define PREFETCHW     prefetcht0
diff --git a/kernel/x86_64/dgemm_tcopy_2.S b/kernel/x86_64/dgemm_tcopy_2.S
index 9881610d7..f8bdbce51 100644
--- a/kernel/x86_64/dgemm_tcopy_2.S
+++ b/kernel/x86_64/dgemm_tcopy_2.S
@@ -39,7 +39,7 @@
 #define ASSEMBLER
 #include "common.h"
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 #define PREFETCHSIZE	16
 #define PREFETCH      prefetcht0
 #define PREFETCHW     prefetcht0
diff --git a/kernel/x86_64/dgemm_tcopy_4.S b/kernel/x86_64/dgemm_tcopy_4.S
index 98ba6473c..6aa83ab46 100644
--- a/kernel/x86_64/dgemm_tcopy_4.S
+++ b/kernel/x86_64/dgemm_tcopy_4.S
@@ -39,7 +39,7 @@
 #define ASSEMBLER
 #include "common.h"
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 #define PREFETCHSIZE	16
 #define PREFETCH      prefetcht0
 #define PREFETCHW     prefetcht0
diff --git a/kernel/x86_64/gemm_kernel_8x4_sse.S b/kernel/x86_64/gemm_kernel_8x4_sse.S
index 1602c13c5..a4ac719d3 100644
--- a/kernel/x86_64/gemm_kernel_8x4_sse.S
+++ b/kernel/x86_64/gemm_kernel_8x4_sse.S
@@ -1713,7 +1713,7 @@
 	ALIGN_4
 
 .L52:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	movss	 1 * SIZE(B), %xmm1
 	movss	 2 * SIZE(B), %xmm2
@@ -1801,7 +1801,7 @@
 	ALIGN_4
 
 .L54:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	movss	 1 * SIZE(B), %xmm1
 
@@ -2689,7 +2689,7 @@
 
 
 .L102:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	movss	 1 * SIZE(B), %xmm1
 	movss	 2 * SIZE(B), %xmm2
@@ -2777,7 +2777,7 @@
 	ALIGN_4
 
 .L104:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	shufps	 $0, %xmm0, %xmm0
 	movaps	%xmm0,  0 * SIZE(BO)
diff --git a/kernel/x86_64/gemm_ncopy_4.S b/kernel/x86_64/gemm_ncopy_4.S
index 7192cecc2..43b286c69 100644
--- a/kernel/x86_64/gemm_ncopy_4.S
+++ b/kernel/x86_64/gemm_ncopy_4.S
@@ -39,7 +39,7 @@
 #define ASSEMBLER
 #include "common.h"
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 #define RPREFETCHSIZE	16
 #define WPREFETCHSIZE (RPREFETCHSIZE * 4)
 #define PREFETCH      prefetcht0
diff --git a/kernel/x86_64/gemm_tcopy_4.S b/kernel/x86_64/gemm_tcopy_4.S
index ba7714b4b..53bad643a 100644
--- a/kernel/x86_64/gemm_tcopy_4.S
+++ b/kernel/x86_64/gemm_tcopy_4.S
@@ -39,7 +39,7 @@
 #define ASSEMBLER
 #include "common.h"
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 #define RPREFETCHSIZE	16
 #define WPREFETCHSIZE (RPREFETCHSIZE * 4)
 #define PREFETCH      prefetcht0
@@ -204,7 +204,7 @@
 	movlps	0 * SIZE(AO4), %xmm3
 	movhps	2 * SIZE(AO4), %xmm3
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO1)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO2)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO3)
@@ -362,7 +362,7 @@
 	movlps	0 * SIZE(AO2), %xmm1
 	movhps	2 * SIZE(AO2), %xmm1
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO1)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO2)
 	PREFETCHW	WPREFETCHSIZE * SIZE(BO)
@@ -381,7 +381,7 @@
 	movsd	2 * SIZE(AO2), %xmm3
 	movhpd	3 * SIZE(AO2), %xmm3
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO1)
 	PREFETCH	RPREFETCHSIZE * SIZE(AO2)
 	PREFETCHW	WPREFETCHSIZE * SIZE(BO)
diff --git a/kernel/x86_64/izamax_sse2.S b/kernel/x86_64/izamax_sse2.S
index c656a652d..36f32b3f0 100644
--- a/kernel/x86_64/izamax_sse2.S
+++ b/kernel/x86_64/izamax_sse2.S
@@ -473,7 +473,7 @@
 	prefetch	PREFETCHSIZE * SIZE(X)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 	prefetchnta	PREFETCHSIZE * SIZE(X)
 #endif
 
diff --git a/kernel/x86_64/symv_L_sse.S b/kernel/x86_64/symv_L_sse.S
index 55780734f..a979049fb 100644
--- a/kernel/x86_64/symv_L_sse.S
+++ b/kernel/x86_64/symv_L_sse.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 12)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 20)
diff --git a/kernel/x86_64/symv_L_sse2.S b/kernel/x86_64/symv_L_sse2.S
index 77331d95f..fc85c98a0 100644
--- a/kernel/x86_64/symv_L_sse2.S
+++ b/kernel/x86_64/symv_L_sse2.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 12)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 20)
diff --git a/kernel/x86_64/symv_U_sse.S b/kernel/x86_64/symv_U_sse.S
index b61182303..a761616ad 100644
--- a/kernel/x86_64/symv_U_sse.S
+++ b/kernel/x86_64/symv_U_sse.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 12)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 20)
diff --git a/kernel/x86_64/symv_U_sse2.S b/kernel/x86_64/symv_U_sse2.S
index 99bc07d50..4901cfa65 100644
--- a/kernel/x86_64/symv_U_sse2.S
+++ b/kernel/x86_64/symv_U_sse2.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 24)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 20)
diff --git a/kernel/x86_64/trsm_kernel_LN_8x4_sse.S b/kernel/x86_64/trsm_kernel_LN_8x4_sse.S
index a31874763..351a9e9fe 100644
--- a/kernel/x86_64/trsm_kernel_LN_8x4_sse.S
+++ b/kernel/x86_64/trsm_kernel_LN_8x4_sse.S
@@ -81,7 +81,7 @@
 #define BORIG	 48(%rsp)
 #define BUFFER	128(%rsp)
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #endif
diff --git a/kernel/x86_64/trsm_kernel_LT_8x4_sse.S b/kernel/x86_64/trsm_kernel_LT_8x4_sse.S
index 887c071ba..712fc9b8c 100644
--- a/kernel/x86_64/trsm_kernel_LT_8x4_sse.S
+++ b/kernel/x86_64/trsm_kernel_LT_8x4_sse.S
@@ -81,7 +81,7 @@
 #define BORIG	 48(%rsp)
 #define BUFFER	128(%rsp)
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #endif
diff --git a/kernel/x86_64/trsm_kernel_RT_8x4_sse.S b/kernel/x86_64/trsm_kernel_RT_8x4_sse.S
index c854b9308..3fda88045 100644
--- a/kernel/x86_64/trsm_kernel_RT_8x4_sse.S
+++ b/kernel/x86_64/trsm_kernel_RT_8x4_sse.S
@@ -81,7 +81,7 @@
 #define BORIG	 48(%rsp)
 #define BUFFER	128(%rsp)
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #endif
diff --git a/kernel/x86_64/zasum_sse2.S b/kernel/x86_64/zasum_sse2.S
index d1e076c31..ca83dca52 100644
--- a/kernel/x86_64/zasum_sse2.S
+++ b/kernel/x86_64/zasum_sse2.S
@@ -249,7 +249,7 @@
 	prefetcht0	PREFETCHSIZE * SIZE(X)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 	prefetchnta	PREFETCHSIZE * SIZE(X)
 #endif
 
diff --git a/kernel/x86_64/zgemm3m_kernel_8x4_sse.S b/kernel/x86_64/zgemm3m_kernel_8x4_sse.S
index a3c69163e..bc81dd688 100644
--- a/kernel/x86_64/zgemm3m_kernel_8x4_sse.S
+++ b/kernel/x86_64/zgemm3m_kernel_8x4_sse.S
@@ -1792,7 +1792,7 @@
 	ALIGN_4
 
 .L52:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	movss	 1 * SIZE(B), %xmm1
 	movss	 2 * SIZE(B), %xmm2
@@ -1880,7 +1880,7 @@
 	ALIGN_4
 
 .L54:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	movss	 1 * SIZE(B), %xmm1
 
@@ -2763,7 +2763,7 @@
 
 
 .L102:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	movss	 1 * SIZE(B), %xmm1
 	movss	 2 * SIZE(B), %xmm2
@@ -2851,7 +2851,7 @@
 	ALIGN_4
 
 .L104:
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 	movss	 0 * SIZE(B), %xmm0
 	shufps	 $0, %xmm0, %xmm0
 	movaps	%xmm0,  0 * SIZE(BO)
diff --git a/kernel/x86_64/zgemm_kernel_4x2_sse.S b/kernel/x86_64/zgemm_kernel_4x2_sse.S
index 7d606aa6f..25c4f33df 100644
--- a/kernel/x86_64/zgemm_kernel_4x2_sse.S
+++ b/kernel/x86_64/zgemm_kernel_4x2_sse.S
@@ -93,7 +93,7 @@
 #define PREFETCHSIZE (16 * 5 + 8)
 #endif
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #define PREFETCHSIZE  160
@@ -1573,7 +1573,7 @@
 	movaps	%xmm14, 24 * SIZE(BO)
 	movaps	%xmm15, 28 * SIZE(BO)
 
-#if defined(PENTIUM4) || defined(GENERIC)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC)
         PREFETCHW	128 * SIZE(BO)
 	PREFETCH	112 * SIZE(B)
 #endif
diff --git a/kernel/x86_64/zgemm_ncopy_2.S b/kernel/x86_64/zgemm_ncopy_2.S
index 845198226..6c56c3618 100644
--- a/kernel/x86_64/zgemm_ncopy_2.S
+++ b/kernel/x86_64/zgemm_ncopy_2.S
@@ -74,7 +74,7 @@
 #define WPREFETCHSIZE 48
 #endif
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 #define RPREFETCHSIZE 32
 #define WPREFETCHSIZE 80
 #endif
@@ -150,7 +150,7 @@
 	movlps	6 * SIZE(AO1), %xmm3
 	movhps	6 * SIZE(AO2), %xmm3
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO2)
 
@@ -191,7 +191,7 @@
 	movsd	6 * SIZE(AO2), %xmm7
 	movhpd	7 * SIZE(AO2), %xmm7
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO2)
 
@@ -295,7 +295,7 @@
 	movapd	%xmm3,   6 * SIZE(B)
 #endif
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO2)
 
diff --git a/kernel/x86_64/zgemm_tcopy_2.S b/kernel/x86_64/zgemm_tcopy_2.S
index 121bbc428..c69872633 100644
--- a/kernel/x86_64/zgemm_tcopy_2.S
+++ b/kernel/x86_64/zgemm_tcopy_2.S
@@ -153,7 +153,7 @@
 	movlps	4 * SIZE(AO2), %xmm3
 	movhps	6 * SIZE(AO2), %xmm3
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO2)
 	prefetcht0	WPREFETCHSIZE * SIZE(BO)
@@ -197,7 +197,7 @@
 	movsd	6 * SIZE(AO2), %xmm7
 	movhpd	7 * SIZE(AO2), %xmm7
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO2)
 	prefetcht0	WPREFETCHSIZE * SIZE(BO)
@@ -316,7 +316,7 @@
 	movlps	4 * SIZE(AO1), %xmm1
 	movhps	6 * SIZE(AO1), %xmm1
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	WPREFETCHSIZE * SIZE(BO)
 #endif
@@ -339,7 +339,7 @@
 	movsd	6 * SIZE(AO1), %xmm3
 	movhpd	7 * SIZE(AO1), %xmm3
 
-#if defined(PENTIUM4) || defined(GENERIC) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(GENERIC) || defined(NANO)
 	prefetcht0	RPREFETCHSIZE * SIZE(AO1)
 	prefetcht0	WPREFETCHSIZE * SIZE(BO)
 #endif
diff --git a/kernel/x86_64/zsum_sse2.S b/kernel/x86_64/zsum_sse2.S
index 6f667164d..bb4225988 100644
--- a/kernel/x86_64/zsum_sse2.S
+++ b/kernel/x86_64/zsum_sse2.S
@@ -219,7 +219,7 @@
 	prefetcht0	PREFETCHSIZE * SIZE(X)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 	prefetchnta	PREFETCHSIZE * SIZE(X)
 #endif
 
diff --git a/kernel/x86_64/zsymv_L_sse.S b/kernel/x86_64/zsymv_L_sse.S
index df190c64c..5618eaccd 100644
--- a/kernel/x86_64/zsymv_L_sse.S
+++ b/kernel/x86_64/zsymv_L_sse.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 24)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 28)
diff --git a/kernel/x86_64/zsymv_L_sse2.S b/kernel/x86_64/zsymv_L_sse2.S
index fa61ac939..05b676ed6 100644
--- a/kernel/x86_64/zsymv_L_sse2.S
+++ b/kernel/x86_64/zsymv_L_sse2.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 24)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 28)
diff --git a/kernel/x86_64/zsymv_U_sse.S b/kernel/x86_64/zsymv_U_sse.S
index 13176ce9c..687f9caa1 100644
--- a/kernel/x86_64/zsymv_U_sse.S
+++ b/kernel/x86_64/zsymv_U_sse.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 24)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 28)
diff --git a/kernel/x86_64/zsymv_U_sse2.S b/kernel/x86_64/zsymv_U_sse2.S
index 1657885c0..110e2b489 100644
--- a/kernel/x86_64/zsymv_U_sse2.S
+++ b/kernel/x86_64/zsymv_U_sse2.S
@@ -63,7 +63,7 @@
 #define PREFETCHSIZE	(16 * 24)
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHW	prefetcht0
 #define PREFETCHSIZE	(16 * 28)
diff --git a/kernel/x86_64/ztrsm_kernel_LN_4x2_sse.S b/kernel/x86_64/ztrsm_kernel_LN_4x2_sse.S
index cd86db289..1eb0bab18 100644
--- a/kernel/x86_64/ztrsm_kernel_LN_4x2_sse.S
+++ b/kernel/x86_64/ztrsm_kernel_LN_4x2_sse.S
@@ -88,7 +88,7 @@
 #define movsd movlps
 #endif
 
-#if defined(PENTIUM4) || defined(CORE2) || defined(PENRYN) || defined(DUNNINGTON) || defined(ATOM) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(CORE2) || defined(PENRYN) || defined(DUNNINGTON) || defined(ATOM) || defined(NANO)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #define PREFETCHNTA  prefetchnta
diff --git a/kernel/x86_64/ztrsm_kernel_LT_4x2_sse.S b/kernel/x86_64/ztrsm_kernel_LT_4x2_sse.S
index 53e5bb7f9..189bd866c 100644
--- a/kernel/x86_64/ztrsm_kernel_LT_4x2_sse.S
+++ b/kernel/x86_64/ztrsm_kernel_LT_4x2_sse.S
@@ -88,7 +88,7 @@
 #define movsd movlps
 #endif
 
-#if defined(PENTIUM4) || defined(CORE2) || defined(PENRYN) || defined(DUNNINGTON) || defined(ATOM) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(CORE2) || defined(PENRYN) || defined(DUNNINGTON) || defined(ATOM) || defined(NANO)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #define PREFETCHNTA  prefetchnta
diff --git a/kernel/x86_64/ztrsm_kernel_RT_4x2_sse.S b/kernel/x86_64/ztrsm_kernel_RT_4x2_sse.S
index 20b93e198..e91401741 100644
--- a/kernel/x86_64/ztrsm_kernel_RT_4x2_sse.S
+++ b/kernel/x86_64/ztrsm_kernel_RT_4x2_sse.S
@@ -88,7 +88,7 @@
 #define movsd movlps
 #endif
 
-#if defined(PENTIUM4) || defined(CORE2) || defined(PENRYN) || defined(DUNNINGTON) || defined(ATOM) || defined(NANO)
+#if defined(PENTIUM4) || defined(PRESCOTT) || defined(CORE2) || defined(PENRYN) || defined(DUNNINGTON) || defined(ATOM) || defined(NANO)
 #define PREFETCH     prefetcht0
 #define PREFETCHW    prefetcht0
 #define PREFETCHNTA  prefetchnta
diff --git a/l1param.h b/l1param.h
index 0b216c7c5..906256499 100644
--- a/l1param.h
+++ b/l1param.h
@@ -31,7 +31,7 @@
 #define movsd	movlps
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define PREFETCH	prefetcht0
 #define PREFETCHSIZE (128 *  10)
 #define FETCH128
diff --git a/l2param.h b/l2param.h
index c5b9d88f5..316d0b5a8 100644
--- a/l2param.h
+++ b/l2param.h
@@ -19,7 +19,7 @@
 #define PREFETCHSIZE	64 * 3
 #endif
 
-#ifdef PENTIUM4
+#if defined(PENTIUM4) || defined(PRESCOTT)
 #define ALIGNED_ACCESS
 #define MOVUPS_A	movaps
 #define MOVUPS_XL	movaps
-- 
2.44.0

